### PR TITLE
PWX-38007: Telemetry cert-secret refresh fix

### PR DESCRIPTION
**Pull request by zoxpx**
* the telemetry needs to force-delete the invalid pure-telemetry-certs secret  (no owner's check)

**What this PR does / why we need it**:

This is a follow-up fix for #1553

**ISSUE**:
* it turns out the `secret/pure-telemetry-certs` may be stored _WITHOUT_ the `metatata.ownerReference`
* note that the px-operator's "normal"  `DeleteSecret()` will remove the secret _only_ if it has required object owners
  - this is not a problem for most of operator-created secrets, since it's the operator itself that creates the `secret` object, and it'll populate the correct `ownerReference`
  - however, with `secret/pure-telemetry-certs` the secret is created by telemetry POD (`ccm-go` container), but owner gets populated at later point in time  (i.e. this particular secret follows 2-step-creation process)

**FIX**:
* as a fix, when we decide we need to remove invalid `secret/pure-telemetry-certs` -- we'll use UNCONDITIONAL `DeleteSecret()` which does not check the object's owners

**Which issue(s) this PR fixes** (optional)
Closes # PWX-38007

**Special notes for your reviewer**:

Manual test:
1. set up INVALID `secret/pure-telemetry-certs` -- which we picked up from a different cluster
   - note, this carries SSL-cert, which has a wrong ClusterUUID embedded
2. made sure to remove the `meradata.ownerReference` on the `secret/pure-telemetry-certs`
3. the `px-operator` now correctly removes the `secret/pure-telemetry-certs` and restarts `pod/px-telemetry-registration-XX`
   - note, BEFORE this fix, the `secret/pure-telemetry-certs` could not get deleted -- it was error-looping since the object's OWNER could not be matched

### CODE REVIEW

File: drivers/storage/portworx/component/telemetry.go [hunk 1]
```diff
@@ -1059,38 +1061,43 @@ func (t *telemetry) validateTelemetrySSLCert(certBytes []byte, cuuid string) err
...
+	if sec == nil {
+		return
+	} else if sec.Name != pxutil.TelemetryCertName {
+		logrus.Errorf("invalid secret name %s/%s  (expected %s)", sec.Namespace, sec.Name, pxutil.TelemetryCertName)
 		return
 	}
-	logrus.Warnf("refreshTelemetrySSLCert - deleting telemetry SSL cert secret %s/%s", dep.Namespace, pxutil.TelemetryCertName)
-	err := k8sutil.DeleteSecret(t.k8sClient, pxutil.TelemetryCertName, dep.Namespace, dep.OwnerReferences...)
+
+	logrus.Warnf("refreshTelemetrySSLCert - deleting telemetry SSL cert secret %s/%s", sec.Namespace, sec.Name)
```
Comment (nikita-bhatia) [inline drivers/storage/portworx/component/telemetry.go:1092]: are we only deleting telemetry cert for refresh? and wit for next reconcilation loop to create it back with owner reference?

**Comment by nikita-bhatia** [comment]
Can we add UTs to cover deletion with different owner reference in this case?

File: drivers/storage/portworx/component/telemetry.go (hunk 1 above)
Comment (zoxpx) [reply drivers/storage/portworx/component/telemetry.go:1092]: The secret is created by `px-telemetry-registration` POD, while the owner-reference is supposed to set up by `px-operator`.  So it's not an "atomic create", and things can break and get interrupted.
* also, in normal circumstances, `px-operator` refuses to delete secrets that don't have the `StorageCluster` set as the owner, probably for added security
* some opportunities where things can get interrupted --
  1. by "install PX" quickly followed by "delete PX"  (<< IMO, common when people install PX for the 1st time) -- or
  2. by cycling `spec.telemetry.enabled: true <-> false`

This particular `refreshTelemetrySSLCert` is ran only when we confirmed that the `secret/pure-telemetry-certs` has the wrong ClusterUUID embedded in the SSL certificate -- so we should absolutely NOT use it with Telemetry
* to make things worse, I think Telemetry backend won't try to match the ClusterUUID from the client SSL-cert, with the actual ClusterUUID that's inside the Telemetry data-payload -- and it'll just accept it, and store it against the WRONG CLUSTER!!  (wrong ClusterUUID)
* so, I have no issues calling "unconditional delete" on the `secret/pure-telemetry-certs` -- when we _know_ it contains WRONG DATA

BTW, once we remove the secret, we'll bounce (remove) the `pod/portworx-telemetry-registration` -- so we don't have to wait for the next reconcilation loop as the deployment will restart the POD.

**Comment by zoxpx** [comment]
> Can we add UTs to cover deletion with different owner reference in this case?

I think it would be an overkill to create mock to emulate broken telemetry registration, especially for this trivial bugfix.
* if you take a look at the https://github.com/libopenstorage/operator/pull/1603/files -- the unit-test coverage gaps are ALL in the error-checking blocks  (i.e. null-checks and such)

**Review by nikita-bhatia — APPROVED** [review]
LGTM, based on discussion, existing UTs are covering main testing areas.

**Comment by zoxpx** [comment]
Thanks for the review Nikita -- merging the PR

**Pull request merged**